src/openzeppelin/account/library.cairo

```diff
@@ -141,12 +142,47 @@ namespace Account:
         return (is_valid=TRUE)
     end
 
+ func is_valid_eth_signature{
+            syscall_ptr : felt*,
+            pedersen_ptr : HashBuiltin*,
+            range_check_ptr,
+            bitwise_ptr: BitwiseBuiltin*
+        }(
+            signature_len: felt,
+            signature: felt*
+        ) -> (is_valid: felt):
+        alloc_locals
+        let (_public_key) = get_public_key()
+        let (__fp__, _) = get_fp_and_pc()
+
+        # This interface expects a signature pointer and length to make
+        # no assumption about signature validation schemes.
+        # But this implementation does, and it expects a the sig_v, sig_r, 
+        # sig_s, and hash elements.
+        let sig_v: felt = signature[0]
+        local sig_r : Uint256 = Uint256(low=signature[1], high=signature[2])
+        local sig_s : Uint256 = Uint256(low=signature[3], high=signature[4])
+        local msg_hash : Uint256 = Uint256(low=signature[5], high=signature[6])
```

Comment:
As far as I see `msg_hash` isn't checked against the passed `call_array`. So an attacker can replay any valid signature with arbitrary `call_array` and `nonce` they'd like to. Can you confirm this? Meanwhile, I'll try to write a proof of concept.